postgres-operator.go

```diff
@@ -57,117 +49,62 @@ func main() {
+	pgoRESTclient := clients.PGORestclient
 
-	operator.Initialize(Clientset)
+	operator.Initialize(kubeClientset)
 
-	namespaceList := ns.GetNamespaces(Clientset, operator.InstallationName)
+	namespaceList := ns.GetNamespaces(kubeClientset, operator.InstallationName)
 	log.Debugf("watching the following namespaces: [%v]", namespaceList)
 
 	//validate the NAMESPACE env var
-	err = ns.ValidateNamespaces(Clientset, operator.InstallationName, operator.PgoNamespace)
+	err = ns.ValidateNamespaces(kubeClientset, operator.InstallationName, operator.PgoNamespace)
 	if err != nil {
 		log.Error(err)
 		os.Exit(2)
 	}
 
-	// start a controller on instances of our custom resource
-	ctx, cancelFunc := context.WithCancel(context.Background())
-
-	pgTaskcontroller := &pgtask.Controller{
-		PgtaskConfig:       config,
-		PgtaskClient:       crdClient,
-		PgtaskScheme:       crdScheme,
-		PgtaskClientset:    Clientset,
-		Queue:              workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
-		Ctx:                ctx,
-		InformerNamespaces: make(map[string]struct{}),
-	}
+	// set up signals so we handle the first shutdown signal gracefully
+	stopCh := signals.SetupSignalHandler()
 
-	pgClustercontroller := &pgcluster.Controller{
-		PgclusterClient:    crdClient,
-		PgclusterScheme:    crdScheme,
-		PgclusterClientset: Clientset,
-		Queue:              workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
-		Ctx:                ctx,
-		InformerNamespaces: make(map[string]struct{}),
-	}
-	pgReplicacontroller := &pgreplica.Controller{
-		PgreplicaClient:    crdClient,
-		PgreplicaScheme:    crdScheme,
-		PgreplicaClientset: Clientset,
-		Queue:              workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
-		Ctx:                ctx,
-		InformerNamespaces: make(map[string]struct{}),
-	}
-	pgPolicycontroller := &pgpolicy.Controller{
-		PgpolicyClient:     crdClient,
-		PgpolicyScheme:     crdScheme,
-		PgpolicyClientset:  Clientset,
-		Ctx:                ctx,
-		InformerNamespaces: make(map[string]struct{}),
-	}
-	podcontroller := &pod.Controller{
-		PodClientset:       Clientset,
-		PodClient:          crdClient,
-		PodConfig:          config,
-		Ctx:                ctx,
-		InformerNamespaces: make(map[string]struct{}),
+	// create a new controller manager with controllers for all current namespaces and then run
+	// all of those controllers
+	controllerManager, err := manager.NewControllerManager(namespaceList)
+	if err != nil {
+		log.Error(err)
+		os.Exit(2)
 	}
-	jobcontroller := &job.Controller{
-		JobConfig:          config,
-		JobClientset:       Clientset,
-		JobClient:          crdClient,
-		Ctx:                ctx,
-		InformerNamespaces: make(map[string]struct{}),
+	controllerManager.RunAll()
+	log.Debug("controller manager created and all included controllers are now running")
+
+	nsKubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClientset, 0)
+	if err != nil {
```

Comment:
This is the same `err` from L78-79. What should be checked after calling `kubeinformers.NewSharedInformerFactory` ?